pkg/resources/task.go

```diff
@@ -191,110 +142,73 @@ func Task() *schema.Resource {
 // ReadTask implements schema.ReadFunc.
 func ReadTask(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
-	taskID, err := taskIDFromString(d.Id())
-	if err != nil {
-		return err
-	}
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
 
-	database := taskID.DatabaseName
-	schema := taskID.SchemaName
-	name := taskID.TaskName
+	taskId := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
 
-	builder := snowflake.NewTaskBuilder(name, database, schema)
-	q := builder.Show()
-	row := snowflake.QueryRow(db, q)
-	t, err := snowflake.ScanTask(row)
-	if errors.Is(err, sql.ErrNoRows) {
+	task, err := client.Tasks.ShowByID(ctx, taskId)
+	if err != nil {
 		// If not found, mark resource to be removed from state file during apply or refresh
 		log.Printf("[DEBUG] task (%s) not found", d.Id())
 		d.SetId("")
 		return nil
 	}
-	if err != nil {
-		return err
-	}
 
-	if err := d.Set("enabled", t.IsEnabled()); err != nil {
+	if err := d.Set("enabled", task.IsStarted()); err != nil {
 		return err
 	}
 
-	if err := d.Set("name", t.Name); err != nil {
+	if err := d.Set("name", task.Name); err != nil {
 		return err
 	}
 
-	if err := d.Set("database", t.DatabaseName); err != nil {
+	if err := d.Set("database", task.DatabaseName); err != nil {
 		return err
 	}
 
-	if err := d.Set("schema", t.SchemaName); err != nil {
+	if err := d.Set("schema", task.SchemaName); err != nil {
 		return err
 	}
 
-	if err := d.Set("warehouse", t.Warehouse); err != nil {
+	if err := d.Set("warehouse", task.Warehouse); err != nil {
 		return err
 	}
 
-	if err := d.Set("schedule", t.Schedule); err != nil {
+	if err := d.Set("schedule", task.Schedule); err != nil {
 		return err
 	}
 
-	if err := d.Set("comment", t.Comment); err != nil {
+	if err := d.Set("comment", task.Comment); err != nil {
 		return err
 	}
 
-	allowOverlappingExecutionValue, err := t.AllowOverlappingExecution.Value()
-	if err != nil {
+	if err := d.Set("allow_overlapping_execution", task.AllowOverlappingExecution); err != nil {
 		return err
 	}
 
-	if allowOverlappingExecutionValue != nil && allowOverlappingExecutionValue.(string) != "null" {
-		allowOverlappingExecution, err := strconv.ParseBool(allowOverlappingExecutionValue.(string))
-		if err != nil {
-			return err
-		}
-
-		if err := d.Set("allow_overlapping_execution", allowOverlappingExecution); err != nil {
-			return err
-		}
-	} else {
-		if err := d.Set("allow_overlapping_execution", false); err != nil {
-			return err
-		}
-	}
-
-	// The "DESCRIBE TASK ..." command returns the string "null" for error_integration
-	if t.ErrorIntegration.String == "null" {
-		t.ErrorIntegration.Valid = false
-		t.ErrorIntegration.String = ""
-	}
-
-	if err := d.Set("error_integration", t.ErrorIntegration.String); err != nil {
+	if err := d.Set("error_integration", task.ErrorIntegration); err != nil {
```

Comment:
is this part of the describe or the show? there was a comment previously that this was part of the describe command. Are we combining the two together?

Reply:
I have removed it because this logic is already in the SDK (mapping task row): 
```
	if r.ErrorIntegration.Valid && r.ErrorIntegration.String != "null" {
		task.ErrorIntegration = r.ErrorIntegration.String
	}
```